### PR TITLE
[ALLUXIO-2451] Add scripts for starting Alluxio in the foreground

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2451

### CODE REVIEW

File: bin/alluxio-proxy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+interacting with the Alluxio filesystem. To configure the proxy, add
+configuration properties in alluxio-site.properties or environment variables in
+alluxio-env.sh."
+
+if [[ "$#" -gt "0" ]]; then
+  echo "${USAGE}"
+  exit 0
+fi
+
+# Log to both the console and the master logs file
```
Comment (apc999) [inline bin/alluxio-proxy]: master -> proxy

File: bin/alluxio-worker [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+the console and default log file. To configure the worker, add configuration
+properties in alluxio-site.properties or environment variables in
+alluxio-env.sh."
+
+if [[ "$#" -gt "0" ]]; then
+  echo "${USAGE}"
+  exit 0
+fi
+
+# Log to both the console and the master logs file
```
Comment (apc999) [inline bin/alluxio-worker]: master -> worker

**Comment by aaudiber** [comment]
@apc999 This is different from `alluxio-start.sh` because it does the minimal amount of work to start the processes, whereas `alluxio-start.sh` does a lot more, e.g. mounting ramfs, killing old processes, and formatting. Most importantly, these scripts run Alluxio in the foreground, unlike `alluxio-start.sh` which cannot run the processes in the foreground because `alluixo-start.sh all` needs to run multiple processes. Running processes in the foreground lets us use console logging, which is important for environments like docker, yarn, and mesos. `alluxio-start.sh` is focused on managing an alluxio cluster, while these scripts are basically just wrappers around running the Alluxio servers.

**Comment by apc999** [comment]
@aaudiber I am still inclined to reuse the previous cli semantics if possible. Adding new shell files is less ideal because (1) they are public APIs which is hard to drop and  (2) too many shell files are hard to maintain too.

I think
```bash
$ bin/alluxio-start.sh -w master
``` 
and
```bash
$ bin/alluxio-start.sh -w proxy
``` 
are already very similar to what `alluxio-master` and `alluxio-proxy` is designed to achieve in this PR, except 
- the logger part which we can definitely probably fix
- ^C quits from the wait not the process which we should probably anyway fix too, as the initial reason to use `-w` is for mesos I think.

In terms of `alluxio-worker`, currently `bin/alluxio-start.sh -w worker` is different as it does format before starting worker, and if "MOPT" is not provided, `SudoMount` is assumed. However, can we simply add something like `SkipMount`? Or what is the default `ALLUXIO_RAMP_FOLDER` for the `alluxio-worker` script in this PR anyway? Can we use that as default?

**Comment by aaudiber** [comment]
(1) The public API surface for these scripts is very small because they take no arguments. Adding this same functionality to alluxio-start.sh would require adding new arguments and functionality, resulting in a larger increase in the API surface.

(2) It's harder to maintain a complicated shell file than a few simple ones. When changing alluxio-start.sh, we have to consider the interactions of the change with the many different options it supports.

(3) The previous CLI isn't failing to achieve a goal due to a bug in the CLI. It just can't achieve the goal because the goal is not compatible with what it already. The goal here is to provide a minimal way to launch an Alluxio process in the foreground. alluxio-start.sh often starts multiple processes at once, so they cannot be in the foreground. I think the only risk to making this change is that users could be confused about what to use. To address this, what do you think of changing the script names to just `master`, `worker`, and `proxy`?

> the logger part which we can definitely probably fix

How can this be fixed in the case of `alluxio-start.sh local -w` or `alluxio-start.sh all -w`? Should we start background processes to simultaneously tail the logs of the master, worker, and proxy? Even then it will be confusing to users that configuring console logging through log4j.properties doesn't generate console logs when using `-w`.

> ^C quits from the wait not the process which we should probably anyway fix too, as the initial reason to use -w is for mesos I think.

Instead of introducing this complexity, why not just run the Alluxio process in the foreground so that it can be managed like a normal application? There are likely to be other issues that arise because `-w` is only pretending to run the application in the foreground, not actually running it in the foreground.

**Comment by jsimsa** [comment]
I agree that it might make sense to put the functionality outside of the alluxio-start.sh script, but I would go one step further and put it into integration/docker/bin.

**Comment by apc999** [comment]
@aaudiber @jsimsa , putting them into `$ALLUXIO_HOME/integration/docker/bin/` works for me. If YARN or mesos could also take advantage from these files, we could also put them to `$ALLUXIO_HOME/integration/bin/`.

For `$ALLUXIO_HOME/bin/`, I am really unsure if we should keep adding new functions into as I view any function this directory as public APIs that we promise to maintain by default. And we should really narrow down our interface. @jsimsa , since you are also the maintainer for public APIs. We should have some guideline on PRs extending public APIs.

**Comment by aaudiber** [comment]
I really don't see why something as simple as "run the application in the foreground" is scary to put into the general-purpose public API, but I'm ok with moving it to the integration public API in `integration/bin`.

**Comment by jsimsa** [comment]
it is not scary, it is confusing for users not familiar with the project ... should I use `alluxio-start.sh master` or should I use `alluxio-master`? ... there should ideally be only entry point (e.g. script) to start a process and options should be used for different modes ... the complexity of internal implementation should not justify confusing API

**Comment by apc999** [comment]
+1 LGTM. As @jsimsa said, having too many choices in `${ALLUXIO_HOME}/bin` may be sometimes confusing for users. Actually due to this PR, I found inconsistency in `alluxio-start.sh` too. I will submit a PR fixing it.

**Pull request merged**